skglm/datafits/single_task.py

```diff
@@ -334,3 +340,14 @@ def full_grad_sparse(
                     grad_j += - X_data[i] * np.sign(tmp) * self.delta
             grad[j] = grad_j / n_samples
         return grad
+
+    def intercept_update_step(self, y, Xw):
+        n_samples = len(y)
+        update = 0.
+        for i in range(n_samples):
+            tmp = y[i] - Xw[i]
```

Comment:
cosmetic comment: `tmp` can be avoided. WDYT of `residual`?

Reply:
It would also be clearer IMO if it was Xw[i] - y[i]   and we removed the minus L350 and L352

Reply:
You're completely right, my bad. Thanks :)